cloud/api-gateway/src/main/java/com/qpay/apigateway/filter/AuthenticationGatewayFilterFactory.java

```diff
@@ -0,0 +1,57 @@
+package com.qpay.apigateway.filter;
+
+import com.qpay.apigateway.exception.MissingTokenException;
+import com.qpay.apigateway.utility.PathsUtils;
+import org.springframework.cloud.gateway.filter.GatewayFilter;
+import org.springframework.cloud.gateway.filter.factory.AbstractGatewayFilterFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.server.reactive.ServerHttpRequest;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.server.ServerWebExchange;
+import reactor.core.publisher.Mono;
+
+@Component
+public class AuthenticationGatewayFilterFactory extends
+        AbstractGatewayFilterFactory<AuthenticationGatewayFilterFactory.Config> {
+
+    private static final String AUTHORIZATION_HEADER = "Authorization";
+
+    private final WebClient webClient;
+
+    public AuthenticationGatewayFilterFactory(final WebClient webClient) {
+        super(Config.class);
+        this.webClient = webClient;
+    }
+
+    @Override
+    public GatewayFilter apply(final Config config) {
+        return (exchange, chain) -> {
+            final var request = exchange.getRequest();
+            if (authMissing(request)) {
+                return onError(exchange, HttpStatus.UNAUTHORIZED);
+            }
+            final var token = request.getHeaders().getOrEmpty(AUTHORIZATION_HEADER).getFirst();
+            webClient.get()
+                    .uri(PathsUtils.VALIDATION_PATH)
+                    .header(AUTHORIZATION_HEADER, token)
+                    .retrieve()
+                    .bodyToMono(Void.class)
+                    .onErrorResume(error -> Mono.error(new MissingTokenException()));
```

Comment:
You have already validated that token exists, it's a mistake to throw here exception with such class naming